### PR TITLE
feat(newserver): Avorion

### LINKED ISSUE #1271: [Server Request] Avorion
Avorion released steam on January 23rd.

Steamdb links:
Game:  https://steamdb.info/app/445220/
Dedicated Server:  https://steamdb.info/app/565060/

Avorion Wiki Entry on setting up dedicated server:  http://wiki.avorion.net/index.php?title=Setting_up_a_server

### PR DESCRIPTION
**Pull request by h3o66**
# Description

Add new Server Avorion

Fixes #1271

## Type of change

* [ ] Bug fix (change which fixes an issue).
* [ ] New feature (change which adds functionality).
* [x] New Server (new server added).
* [ ] Refactor (restructures existing code).
* [ ] Comment update (typo, spelling, explanation, examples, etc).

## Checklist

PR will not be merged until all steps are complete.

* [x] This pull request links to an issue.
* [x] This pull request uses the `develop` branch as its base.
* [x] This code follows the style guidelines of this project.
* [x] I have provided Co-author details below.
* [x] I have performed a self-review of my own code.
* [x] I have checked that this code is commented where required.
* [x] I have provided a detailed enough description of this PR.
* [ ] I have checked If documentation needs updating.

## Provide Github Email

Fill out below info or tick box below:
```
Co-authored-by: Christian Birk <chris.h3o66@gmail.com>
```

- [ ] I do not wish to provide an email. I am aware this will hide me as the author of this commit.

All pull requests will now be squashed to create a tidy commit history and simplify changelog creation. You can provide either your own email or a GitHub-provided no-reply email.

When a PR is squashed the author becomes the person who squashed the PR. This removes you as the author of your own PR.
The only workaround for this is to add your details as a co-author. More info about co-authors can be found [here](https://help.github.com/en/articles/creating-a-commit-with-multiple-authors).

## Documentation

If documentation does need updating either update it by creating a PR (preferred) or request a documentation update.
* User docs: https://github.com/GameServerManagers/LinuxGSM-Docs
* Dev docs: https://github.com/GameServerManagers/LinuxGSM-Dev-Docs

**Thank you for your Pull Request!**

### CODE REVIEW

**Comment by h3o66** [comment]
> seems you must run this server for once in order to create config file otherwise it would say config file not found?

yeah i intended this for now like this, but i could do a "fix" to start the server once after the installation to let the server create the config here

**Comment by Frisasky** [comment]
> 
> 
> > seems you must run this server for once in order to create config file otherwise it would say config file not found?
> 
> yeah i intended this for now like this, but i could do a "fix" to start the server once after the installation to let the server create the config here

the fix script for kf2 do exact same thing, so just copy and paste was enough i think

another problem is the server did not work with new resource usage monitor function, install with latest develop codebase gives me ps command error
https://github.com/GameServerManagers/LinuxGSM/pull/2527

**Comment by h3o66** [comment]
since this server has its own init function I used it in the postinstall fix
If someone else wants to test it, feel free to do so.

**Comment by dgibbs64** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/LinuxGSM/LinuxGSM/pullRequest?prid=4018261&bid=13900795)

**Comment by dgibbs64** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/LinuxGSM/LinuxGSM/pullRequest?prid=4018261&bid=13900795)

**Comment by dgibbs64** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/LinuxGSM/LinuxGSM/pullRequest?prid=4018261&bid=13900795)

**Comment by dgibbs64** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/LinuxGSM/LinuxGSM/pullRequest?prid=4018261&bid=13900795)

**Comment by Frisasky** [comment]
does this server currently broken? as long as i join the server the server crashed with lua error

**Comment by Frisasky** [comment]
> 
> 
> I guess this will break the parameter passed for the Postinstall , as these are then seen as one parameter.
> Did you test this ?

without the quotes the server still crashed

**Comment by dgibbs64** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/LinuxGSM/LinuxGSM/pullRequest?prid=4018261&bid=13900795)

**Comment by h3o66** [comment]
Will review this once more when i have time

**Comment by dgibbs64** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GameServerManagers/LinuxGSM/pullRequest?prid=4018261&bid=13900795)

**Comment by dgibbs64** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GameServerManagers/LinuxGSM/pullRequest?prid=4018261&bid=13900795)

**Comment by h3o66** [comment]
@Frisasky looks good now.
At least with my test it works now

**Comment by dgibbs64** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GameServerManagers/LinuxGSM/pullRequest?prid=4018261&bid=13900795)

**Comment by dgibbs64** [comment]
found bug that prevents starting the executable unless you are in the directory above. This goes against how all other servers work.

The fix is to create a symbolic link in the bin directory to the data directory. This bug should be raised with the developers of the game. @h3o66 If possible please can you email the developers to let them know the bug so they can resolve it. I will update the fix to get the server working. The fix can be removed in the future should they fx the issue

File: lgsm/functions/info_messages.sh [hunk 1]
```diff
@@ -636,6 +636,20 @@ fn_info_message_ark(){
 	} | column -s $'\t' -t
 }
 
+fn_info_message_avorion() {
```
Comment (dgibbs64) [inline lgsm/functions/info_messages.sh]: ports need clarifying

File: lgsm/functions/info_messages.sh [hunk 2]
```diff
@@ -541,7 +541,7 @@ fn_info_message_ports(){
 
 	parmslocation="${red}UNKNOWN${default}"
 	# engines/games that require editing in the config file
-	local ports_edit_array=( "avalanche2.0" "avalanche3.0" "Ballistic Overkill" "dontstarve" "Eco" "idtech2" "idtech3" "idtech3_ql" "lwjgl2" "Minecraft Bedrock" "Project Cars" "projectzomboid" "quake" "refractor" "realvirtuality" "renderware" "seriousengine35" "Stationeers" "teeworlds" "terraria" "unreal" "unreal2" "unreal3" "TeamSpeak 3" "Mumble" "7 Days To Die" "wurm")
+	local ports_edit_array=( "avalanche2.0" "avalanche3.0" "Avorion" "Ballistic Overkill" "dontstarve" "Eco" "idtech2" "idtech3" "idtech3_ql" "lwjgl2" "Minecraft Bedrock" "Project Cars" "projectzomboid" "quake" "refractor" "realvirtuality" "renderware" "seriousengine35" "Stationeers" "teeworlds" "terraria" "unreal" "unreal2" "unreal3" "TeamSpeak 3" "Mumble" "7 Days To Die" "wurm")
```
Comment (dgibbs64) [inline lgsm/functions/info_messages.sh]: Clarify if ports are edited in the config of parms

File: lgsm/functions/info_messages.sh (hunk 1 above)
Comment (h3o66) [reply lgsm/functions/info_messages.sh:639]: well, the rcon port for example is only enabled if you set a password in the config file.
As default it is in serverfiles/galaxy/LinuxGSM/server.ini

example with rcon enabled:
```
tcp        0      0 1.2.3.4:27015   0.0.0.0:*               LISTEN      20360/./bin/Avorion 
udp        0      0 1.2.3.4:27008   0.0.0.0:*                           20360/./bin/Avorion 
udp        0      0 1.2.3.4:27020   0.0.0.0:*                           20360/./bin/Avorion 
```

from the log of the server (located in serverfiles/galaxy/LinuxGSM/):
```
Game Port: 27000
Steam Port: 27021
Steam Query Port: 27020
Query Port: 27008
```

if disabled, it is also noted there in the log:
`Warning: No RCON password set. RCON disabled.`

Comment (h3o66) [reply lgsm/functions/info_messages.sh:639]: anyway the 3rd gameport does not seem to be right anymore, i will exchnage this one with the queryport.

**Comment by h3o66** [comment]
> found bug that prevents starting the executable unless you are in the directory above. This goes against how all other servers work.
> 
> The fix is to create a symbolic link in the bin directory to the data directory. This bug should be raised with the developers of the game. @h3o66 If possible please can you email the developers to let them know the bug so they can resolve it. I will update the fix to get the server working. The fix can be removed in the future should they fx the issue

well - or in the end we just use the provided script and then we do not need to use the hacky symlink.
Changed that with the latest commits.

**Pull request merged**